### ISSUE TITLE
Train arrival time algorithm

### ISSUE THREAD

**Issue by jrootham** [CivicTechTO/ttc_subway_times#40]
Start by selecting all the samples for a train ordered by timestamp.
Merge all samples for a train in the same sample period to get a good estimate of the location.
Use those location estimates to generate arrival time estimates.

**Comment by radumas** [comment]
Hi @jrootham,

You can have at look at https://github.com/CivicTechTO/ttc_subway_times/issues/13 for previous work on this. As well as [`doc/filtering_observed_arrivals.ipynb`](https://github.com/CivicTechTO/ttc_subway_times/blob/master/doc/filtering_observed_arrivals.ipynb)

**Comment by radumas** [comment]
I realize this isn't really written out in plain English in the notebook. The procedure I tried was:
1. Filter observations where the train is "At Station" or is <1 minute from arriving.
2. Order those observations by expected arrival time (`create_date` + `timint`)
3. If the `traindirection` is the same as the previous observation for this train, then it's the same trip. If if changes, then it's a different trip.
4. For each trip, the minimum estimated arrival is the arrival time, and the maximum estimated arrival is the "departure time"

Results.... could be improved upon.

**Comment by jrootham** [comment]
Hi Raphael

I think we should talk about this tonight.

Jim
